lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -362,4 +362,20 @@ sub setup_vm_simple_dns_with_ip {
     record_info("Simple DNS setup in /etc/hosts for $_ip $_vm is successful!", script_output("cat /etc/hosts"));
 }
 
+sub get_guest_status {
+    my ($guest, %args) = @_;
+    my $timeout = $args{timeout} // "180";
+    #Ensure the given guest as running status
+    if (script_run("virsh list --all | grep $guest | grep running") ne 0) {
+        assert_script_run "virsh list --all | grep $guest";
+        save_screenshot;
+        die "Error: $guest should keep running, please check manually!";
+    } else {
+        #Ensure the SSH connection for the given guest
+        die "Error: SSH $guest failed, please check manually!" if (script_retry("nmap $guest -PN -p ssh | grep open", delay => 30, retry => 6, timeout => $timeout) ne 0);
+        #Ensure the ICMP PING responses for the given guest
+        die "Error: Ping $guest failed, please check manually!" if (script_retry("ping -c5 $guest", delay => 30, retry => 6, timeout => $timeout) ne 0);
+    }
```

Comment:
From new development, I think we can start trying nc instead of nmap, because nc is installed by default, for example, https://www.moreofless.co.uk/netcat-nc-port-scanning

Reply:
namp is more powerful, but for ssh port detecting, I think nc might be enough.

Reply:
yes, after do more investigation, figure out nc(netcat) is simple and easy tool to detect SSH port for given guest system. next, try to use nc to detect SSH instead of namp tool. meanwhile, do more local validation runs.